### PR TITLE
[SD-3654] Display the history of published sluglines

### PR DESCRIPTION
**Pull request by marwoodandrew**

### CODE REVIEW

File: server/apps/desks.py [hunk 1]
```diff
@@ -255,3 +260,94 @@ def get(self, req, lookup):
...
+
+    url = 'desks/<regex("[a-f0-9]{24}"):desk_id>/sluglines'
+    datasource = {'source': 'published',
+                  'search_backend': 'elastic',
+                  'default_sort': [('place.name', 1), ('slugline', 1)],
+                  'elastic_filter': {"and": [{"range": {"versioncreated": {"gte": "now-2400H"}}},
+                                             {"term": {"last_published_version": True}},
+                                             {"term": {"type": "text"}}
+                                             ]}}
+    resource_methods = ['GET']
```
Comment (sivakuna-aap) [inline server/apps/desks.py]: `item_methods = []` is missing.

File: client/app/scripts/superdesk-desks/views/slugline-items.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<div ng-if="loading" class="loading"></div>
+
+<div ng-if="!loading && !items.length" class="no-items" translate>No items with this published</div>
+
+<ul ng-if="!loading && items.length" class="inline-content-items content-tasks" >
```
Comment (sivakuna-aap) [inline client/app/scripts/superdesk-desks/views/slugline-items.html]: `track by` and `one time bindings`

File: client/app/scripts/superdesk-desks/desks.js [hunk 3]
```diff
@@ -368,6 +368,26 @@
...
+            },
+            link: function(scope, elem) {
+                scope.items = [];
+                scope.loading = true;
+                api.get('desks/' + scope.desk + '/sluglines').then(function(items) {
+                    scope.items = items._items;
+                    scope.loading = false;
+                }, function() {
+                    scope.loading = false;
+                });
```
Comment (plamut) [inline client/app/scripts/superdesk-desks/desks.js]: The `scope.loading` flag can be cleared in `finally()`, avoiding repetition. The error handler can then be dropped as well.

(although it's relatively minor since the logic is so simple)

File: client/app/scripts/superdesk-desks/views/slugline-items.html (hunk 2 above)
Comment (plamut) [inline client/app/scripts/superdesk-desks/views/slugline-items.html]: +1, please check where these can be applied in the code below.

File: client/app/scripts/superdesk-desks/views/slugline.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<div class="desk-header">
+	<div class="desk-name pull-left">{{ desk.name }} 
+		<span class="view" ng-if="isMemberOf(desk)" ng-click="openDeskView(desk, 'content')" title="{{ 'Open desk content' | translate}}"><i class="icon-external"></i></span>
+	</div>
+	<div class="desk-action pull-right dropdown" dropdown ng-if="privileges.desks">
+		<button class="dropdown-toggle" dropdown-toggle><i class="icon-dots"></i></button>
+		<ul class="dropdown-menu">
+			<li><button ng-click="openDesk('general', desk)"><i class="icon-pencil"></i> {{ 'Edit desk' | translate }}</button></li>
```
Comment (plamut) [inline client/app/scripts/superdesk-desks/views/slugline.html]: One-time binding should be used for the fixed string.

**Comment by plamut** [comment]
Generally good IMO, just needs a few final touches. Also please check if the tests indeed cover the edge cases, too, it seems to me that only the main cases are tested?
(I might be wrong though...)

**Comment by ioanpocol** [comment]
PR https://github.com/superdesk/superdesk/pull/1408 has fixed the e2e tests problems.
Please rebase.

**Comment by ioanpocol** [comment]
:+1:
